ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -748,10 +762,42 @@ static void HandleKernelRequest(void* messageMemory, uint32_t messageSize)
             break;
         }
         
+        case MessageType_KtoU_NotifyFileHardLinkCreated:
+        {
+#if DEBUG
+            // TODO(Mac): Move the following line out of the DEBUG block once we actually need the information. Currently just causes warning-as-error in release build.
+            const char* relativeFromPath = GetRelativePath(request.paths[MessagePath_From], s_virtualizationRootFullPath.c_str());
+
+            cout << "PrjFSLib.HandleKernelRequest: hard-linked " << request.paths[MessagePath_From] << " -> " << absolutePath << " (absolute), ";
+            if (relativeFromPath != nullptr)
+            {
+                cout << "from this root (relative path " << relativeFromPath << ") ";
+            }
+            if (relativePath != nullptr)
+            {
+                cout << "into this root (relative path " << relativePath << ")";
+            }
+            cout << endl;
+#endif
+            
+            bool isDirectory = requestHeader->messageType == MessageType_KtoU_NotifyDirectoryRenamed;
```

Comment:
There is no need to check `requestHeader->messageType == MessageType_KtoU_NotifyDirectoryRenamed` because in this case we're handing `MessageType_KtoU_NotifyFileHardLinkCreated`

Reply:
Thanks, missed that leftover when I removed the "from" provider messaging for on rename.